daffodil-runtime1/src/main/java/org/apache/daffodil/api/validation/ValidationException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.api.validation;
+
+public interface ValidationException extends ValidationFailure {
```

Comment:
We should rename this to ValidationError since ValidationException as a name tends to imply it is derived from Exception or Runtime Exception, but it isn't.

Reply:
We already have a ValidationError class in SDE. Won't renaming it to that cause confusion between the two? I think it's called Exception because it is backed by a throwable?

Reply:
JAPI/SAPI doesn't offer it, so I've moved it out of the API